### DISCUSSION TITLE
Framerate presets, Resolution presets

### DISCUSSION THREAD

**Discussion by axiomgraph** [friction2d/friction#88] [Ideas]
I think it might be useful if there are framerate presets (24, 25, 30, 59.94, and 60) and resolution presets (720p, 1080p, 2K, and 4K) in the scene properties window. so users don't need to find these values by themselves. and adding an option for uniform scaling in transform.

**Comment by AlexKiryanov** [comment]
Hi all! As an addition, it would be great if wee could also change the aspect of the pixel (so you could select not only a square pixel)

**Reply by axiomgraph** [reply]
that would be useful too!

**Comment by rodlie** [comment]
I don't have any need for fps/res presets, but why not....

What should be added?

- 23.97 FPS
- 24 FPS
- 25 FPS
- 30 FPS
- 50 FPS
- 59.94 FPS
- 60 FPS
- 120 FPS
- 240 FPS

more?

and what about resolutions, there are many :)

* 16:9
    * 3840x2160 (4K)
    * 2560x1440 (QHD)
    * 1920x1080 (Full HD)
    * 1280x720 (720p)
    * 854x480 (480p)
    * 640x360 (360p)
    * 5120x2880 (5K)
    * 7680x4320 (8K)
    * 15360x8640 (16K)
    * 1024 × 576 (WSVGA)
* 18.5:9
    * 3840x2048 (UHD+)
* 21:9
    * 3440x1440 (UWQHD)
    * 2560x1080 (WFHD)
    * 5120x2160 (5K UWQHD)
* 2.35:1
    * 3996x1680 (2K)
    * 2048x858 (2K)
    * 1920x800 (HD+)
* 4:3
    * 1024x768 (XGA)
    * 800x600 (SVGA)
    * 640x480 (VGA)
    * 1600x1200 (UXGA)
    * 2048x1536 (QXGA)
    * 1280 × 960 (QuadVGA)
* 1:1
    * 1080x1080
    * 512x512
    * 256x256
    * 2048x2048
    * 4096x4096
* DCI
    * 2K DCI: 2048 × 1080 (1.9:1 aspect ratio)
    * 4K DCI: 4096 × 2160 (1.9:1 aspect ratio)
    
more?

**Reply by axiomgraph** [reply]
For fps presets May be up to 60fps is enough. For resolution presets 16:9 resolutions, vertical resolutions  and other resolutions that mostly use in production. user ability to save his own resolutions as new presets so he can use in future.

**Comment by rodlie** [comment]
So, I added basic preset support:
![friction-res-preset](https://github.com/friction2d/friction/assets/34516798/b010dc41-0bcb-4a4b-a196-6d5adf765644)

![friction-fps-preset](https://github.com/friction2d/friction/assets/34516798/d93b5374-f2d8-47f1-a78c-4233f6cf70f8)

It will auto add wxh/fps to presets if not already present. If you want other behavior let me know.

There is currently no UI to remove presets. Will add something to preferences sometime this week.

Default presets are:

### Resolution
* 1280x720
* 1920x1080
* 2560x1440
* 3840x2160

### FPS
* 24
* 25
* 30
* 50
* 60

If anyone wants any changes to the defaults let me know.

**Reply by axiomgraph** [reply]
Appreciate your efforts thanks! will come with more ideas if that speed up workflow. one thing I have in mind is that if a user imports a video then friction will show a dialog that asking using video frame rate for the project. may be helpful when using videos in projects.
